### PR TITLE
Rollback User Creation on MailException

### LINKED ISSUE #85: User creation not rolled back after sending the activation mail failed
When the server fails to send the activation mail, the user creation is not rolled back. When trying to register the same user after a failed attempt, the user name is already taken. The user creation has to be rolled back after an error occurs.

### PR DESCRIPTION
**Pull request by sputnik27**
Fixes #85

### CODE REVIEW

**Comment by wallner** [comment]
i would like to see some unit tests, please.

File: src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java [hunk 1]
```diff
@@ -92,9 +94,11 @@ public String register(@Valid final RegistrationUser registrationUser,
             return "registration";
         }
 
-        user = registrationService.saveRegistrationUser(user);
-
-        registrationService.sendRegistrationEmail(user, request);
+        try {
+            registerUserOrRollback(user, request);
+        } catch (RollbackUserCreationException e) {
+            return "registration";
```
Comment (dacrome) [inline src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java]: An error message should be returned, like [here](https://github.com/sputnik27/addon-self-administration/blob/rollback-user-creation-after-sending-mail-failed/src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java#L91).

**Comment by sputnik27** [comment]
I have amended the commit - it includes tests now and fixes the issue that @dacrome mentioned.

File: src/main/java/org/osiam/addons/self_administration/SelfAdministration.java [hunk 2]
```diff
@@ -89,4 +87,14 @@ public Filter characterEncodingFilter() {
         characterEncodingFilter.setForceEncoding(true);
         return characterEncodingFilter;
     }
+
+    @Bean
+    public OsiamConnector osiamConnector(Config config) {
+        OsiamConnector.Builder oConBuilder = new OsiamConnector.Builder().
+                setAuthServerEndpoint(config.getAuthServerHome()).
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/SelfAdministration.java]: Coding conventions:

> When an expression will not fit on a single line, break it according to these general principles:
> [...]
> * Break before an operator.
>
> [...]

File: src/main/java/org/osiam/addons/self_administration/controller/AccountManagementService.java [hunk 3]
```diff
@@ -133,9 +132,7 @@ public void deleteUser(String userId, AccessToken token) {
      *            the access token
      */
     private User getUser(String userId, AccessToken token) {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/controller/AccountManagementService.java]: Please, inline this method.

File: src/main/java/org/osiam/addons/self_administration/exception/RollbackUserCreationException.java [hunk 4]
```diff
@@ -0,0 +1,4 @@
+package org.osiam.addons.self_administration.exception;
+
+public class RollbackUserCreationException extends OsiamException {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/exception/RollbackUserCreationException.java]: Unused class. Please delete.

File: src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java [hunk 5]
```diff
@@ -74,7 +75,7 @@ public String getRegistrationForm(final Model model) {
 
     @RequestMapping(method = RequestMethod.POST)
     public String register(@Valid final RegistrationUser registrationUser,
-            final BindingResult bindingResult, final Model model, final HttpServletRequest request) {
+                           final BindingResult bindingResult, final Model model, final HttpServletRequest request) {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java]: wrong indentation

**Comment by tkrille** [comment]
Unused field: https://github.com/osiam/addon-self-administration/pull/138/files#diff-12276534a6b2f79a074ca9834b34a50bR56

File: src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java [hunk 6]
```diff
@@ -110,19 +117,16 @@ public String register(@Valid final RegistrationUser registrationUser,
...
-     *            the user's activation token, send by E-Mail
-     *
+     * @param userId          the id of the registered user
+     * @param activationToken the user's activation token, send by E-Mail
      * @return the registrationSuccess page
      */
     @RequestMapping(value = "/activation", method = RequestMethod.GET)
     public String confirmation(@RequestParam("userId") final String userId,
-            @RequestParam("activationToken") final String activationToken, final Model model) {
+                               @RequestParam("activationToken") final String activationToken, final Model model) {
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/registration/RegistrationController.java]: Wrong indentation

File: src/main/java/org/osiam/addons/self_administration/registration/RegistrationService.java [hunk 7]
```diff
@@ -116,6 +93,7 @@ public void sendRegistrationEmail(User user, HttpServletRequest request) {
 
         Locale locale = SelfAdministrationHelper.getLocale(user.getLocale());
 
+
```
Comment (tkrille) [inline src/main/java/org/osiam/addons/self_administration/registration/RegistrationService.java]: unneeded empty line

**Pull request merged**